util/src/math.rs

```diff
@@ -0,0 +1,25 @@
+pub trait Log2 {
+    /// Returns the log base 2 of `self`.
+    fn log2(self) -> Self;
+}
+
+impl Log2 for usize {
+    /// Fast log base 2 implementation for integers.
```

Comment:
Perhaps we should make sure to say this is log2, rounding up?

FWIW I think log2 rounding down can be a bit more efficient (though also a bit more gross, at least in my first pass, and only if it works for our use-case here) due to there basically being an instruction for that. For example an implementation like this one comes out to only a couple of instructions:

```rust
pub const fn log2_floor(u: usize) -> usize {
  const WIDTH: usize = mem::size_of::<usize>() * 8;
  WIDTH - (u | 1).leading_zeros() as usize - 1
}
```

```asm
example::log2_floor:
        or      rdi, 1
        bsr     rax, rdi
        ret
```
https://godbolt.org/z/r6oWqK

Reply:
yeah, the buddy block algorithm relies on this being `ceil(log2(x))`; this actually caused a bug when i used a fast impl that rounded down. i'll rename the function.